tinygrad/tensor.py

```diff
@@ -322,6 +322,15 @@ def numpy(self) -> 'np.ndarray':  # type: ignore [name-defined] # noqa: F821
     assert all_int(self.shape), f"no data if shape is symbolic, {self.shape=}"
     return np.frombuffer(self._data(), dtype=_to_np_dtype(self.dtype)).reshape(self.shape)
 
+  def clone(self) -> Tensor:
+    """
+    Creates a clone of this tensor allocating a seperate buffer for the data.
+    """
+    ret = Tensor(self.lazydata.clone(), self.device, requires_grad=self.requires_grad)
+    if self.grad is not None: ret.grad = self.grad.to(self.device)
+    if hasattr(self, '_ctx'): ret._ctx = self._ctx
```

Comment:
Careful, why do you want to do this? Clone should be at the Function level

Reply:
I switched it to `self.grad.clone()` instead, but don't actually need it in my case just thought it would be a good addition.

Why would I need this at the function level when clone is only doing a data copy? It is not doing anything more than the `Tensor.to` method does and uses the same boundaries.